### ISSUE TITLE
Squiz.Scope.StaticThisUsage incorrectly looking inside closures

### ISSUE THREAD

**Issue by WarriorXK** [squizlabs/PHP_CodeSniffer#1747]
Currently the StaticThisUsageSniff shows an error in the following case:

```php
<?php

class A {
    static public function Test() {
        return function() {
            echo($this->_name . PHP_EOL);
        };
    }
}

class B {

    protected $_name = 'Bob';

}

$closure = A::Test();
$b = new B();

$closure = $closure->bindTo($b, B::class);
$closure();
```

The sniff should either not error or downgrade it to a warning if the $this reference is inside a closure.

**Comment by gsherwood** [comment]
No error should be reported - the sniff wasn't skipping the closure like it should have been. I think it might actually pre-date closures. It was a bit of a mess too, so I cleaned it up.

Thanks for reporting this.
